packages/truffle-contract-sources/index.js

```diff
@@ -16,5 +18,9 @@ module.exports = (pattern, callback) => {
     follow: true // follow symlinks
   };
 
-  glob(pattern, globOptions, callback);
+  if (callbackPassed) {
+    glob(pattern, globOptions, callback);
+  } else {
+    return promisifiedGlob(pattern, globOptions);
+  }
```

Comment:
And here, use the promisified version for both, and do the callback string-up with a `.then`

Reply:
So like the following?
```
return promisify(glob(pattern, globOptions)
  .then(files => {
    callback(null, files);
  })
  .catch(error => {
    callback(error);
  });
} else {
  return promisify(glob(patter, globOptions);
}
```
Is that what you mean?

Reply:
Oh I see, you mean remove the if and just do...
```
return promisify(glob(patther, globOptions))
  .then(files => {
    if (callbackPassed) callback(null, files);
    return files;
  .catch(error => {
    if (callbackPassed) callback(error);
    throw error;
  });
```
Yeah?

Reply:
Either one!